### PR TITLE
Read lastLogin time of ldap account suspension notifier in the configured pattern

### PR DESCRIPTION
**Pull request by ayshsandu**
This PR has a dependency to https://github.com/wso2/carbon-kernel/pull/2702.

**Hence, merge only when carbon-kernel is released** with above PR

### CODE REVIEW

File: components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java [hunk 1]
```diff
@@ -145,4 +148,45 @@ public void init(RealmConfiguration realmConfiguration) {
...
     }
 
+    /* Convert Active Directory date format (Generalized Time) to WSO2 format.
+     *
+     * @param date Date formatted in Active Directory date format.
+     * @return Date formatted in WSO2 date format.
+     */
+    private long convertDateFormat(String date) {
+
+        //If the user-store uses a different timestamp than WSO2 format
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java:158]: Shall we add a space after `//` and end the sentence with a `.`? Shall we edit in other relevant places as well?

File: components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java [hunk 2]
```diff
@@ -145,4 +148,45 @@ public void init(RealmConfiguration realmConfiguration) {
...
+        String dateTimeFormat = realmConfiguration.getUserStoreProperty(UserStoreConfigConstants.dateAndTimePattern);
+        if(StringUtils.isNotEmpty(dateTimeFormat)){
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(dateTimeFormat);
+            OffsetDateTime offsetDateTime = OffsetDateTime.parse(date, dateTimeFormatter);
+            Instant instant = offsetDateTime.toInstant();
+            return instant.toEpochMilli();
+        }
+        return  Long.parseLong(date.toString());
+    }
+
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java:168]: Shall we remove this extra new line?

File: components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java [hunk 3]
```diff
@@ -145,4 +148,45 @@ public void init(RealmConfiguration realmConfiguration) {
         return users;
     }
 
+    /* Convert Active Directory date format (Generalized Time) to WSO2 format.
+     *
+     * @param date Date formatted in Active Directory date format.
+     * @return Date formatted in WSO2 date format.
+     */
+    private long convertDateFormat(String date) {
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java:156]: IMO, this method name should be more self-explanatory. Ex:
- getDateInWSO2Format
- convertToWSO2DateFormat

File: components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java [hunk 4]
```diff
@@ -145,4 +148,45 @@ public void init(RealmConfiguration realmConfiguration) {
...
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(dateTimeFormat);
+            OffsetDateTime offsetDateTime = OffsetDateTime.parse(date, dateTimeFormatter);
+            Instant instant = offsetDateTime.toInstant();
+            return instant.toEpochMilli();
+        }
+        return  Long.parseLong(date.toString());
+    }
+
+
+    protected String getSearchFilter(long lookupMin, long lookupMax, String lastLoginTimeAttribute) {
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java:170]: Shall we add method comments here?

File: components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java (hunk 1 above)
Comment (ayshsandu) [reply components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/ldap/LDAPNotificationReceiversRetrieval.java]: fixed

**Comment by pulasthi7** [comment]
PR builder : https://github.com/wso2/product-is/actions/runs/359150914

**Pull request merged**